clippy_lints/src/formatting.rs

```diff
@@ -242,26 +243,36 @@ fn has_unary_equivalent(bin_op: BinOpKind) -> bool {
     bin_op == BinOpKind::And || bin_op == BinOpKind::Mul || bin_op == BinOpKind::Sub
 }
 
+fn indentation(cx: &EarlyContext<'_>, span: Span) -> Option<u64> {
+    let lo = cx.sess.source_map().lookup_char_pos(span.lo());
+    lo.file
+        .get_line(lo.line - 1 /* line numbers in `Loc` are 1-based */)
+        .map(|line| line[..lo.col.0].chars().map(|c| if c == '\t' { 8 } else { 1 }).sum())
```

Comment:
I wouldn't special case tab chars here. Since tab chars are always formatted differently, adding 8 for tabs can be correct in one editor, but wrong in others.

Reply:
I know, but 8 is de-facto default tab size, and it's IMO it's better than not accounting for it at all (or maybe better solution is to not lint at all if tab chars are present?)

Reply:
I think we should lint, when tabs are present, since formatting something like

```
    1
        - 2,
// the same as 
    1
    - 2,
// in some other editor
```

with tabs and then rely on the formatting is weird. 

Also if someone uses tabs instead of spaces, they probably use it in every line, so it doesn't have to be special cased.

What I'm saying is, that the biggest argument against tabs is, that the formatting is inconsistent across editors. This is a heuristic that relies on consistent formatting. So I wouldn't try to solve the tab-formatting problem at all here.

Reply:
I special cased it because of 
```
1, 2, 3 -
            4,
 5, 6
```
(although I don't know if that's a possible rustfmt output).
Well worst case scenario if I we don't special case it we get a false-positive, so I'll remove it.

Reply:
You can configure rustfmt to use [hard_tabs](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#hard_tabs). But then rustfmt will use tabs for indentation and spaces for alignment. So also in this case special casing tabs isn't necessary.